src/vector_index.c

```diff
@@ -100,17 +101,24 @@ IndexIterator *NewVectorIterator(RedisSearchCtx *ctx, VectorQuery *vq, IndexIter
       if ((err = VecSimIndex_ResolveParams(vecsim, vq->params.params, array_len(vq->params.params),
                                            &qParams)) != VecSim_OK) {
         err = VecSimResolveCode_to_QueryErrorCode(err);
-        QueryError_SetErrorFmt(status, err, "Error parsing vector similarity parameters: %s",
+        QueryError_SetErrorFmt(q->status, err, "Error parsing vector similarity parameters: %s",
                                QueryError_Strerror(err));
         return NULL;
       }
       if (expBlobSize(vecsim) != vq->knn.vecLen) {
-        QueryError_SetErrorFmt(status, QUERY_EINVAL,
+        QueryError_SetErrorFmt(q->status, QUERY_EINVAL,
                                "Error parsing vector similarity query: query vector blob size (%zu) does not match index's expected size (%zu).",
                                vq->knn.vecLen, expBlobSize(vecsim));
         return NULL;
       }
-      return NewHybridVectorIterator(vecsim, vq->scoreField, vq->knn, qParams, child_it);
+      HybridIteratorParams hParams = {.index = vecsim,
+                                      .query = vq->knn,
+                                      .qParams = qParams,
+                                      .vectorScoreField = vq->scoreField,
+                                      .ignoreDocScore = q->opts->flags & Search_IgnoreScores,
+                                      .childIt = child_it
+      };
+      return NewHybridVectorIterator(hParams);
```

Comment:
nitpick, I do not see why it is better to wrap it with a struct, you could have pass all those argument to the function and use new lines and indentations to make it look almost the same as it look now. It could be useful if you would have initialised it somewhere else and then use it here...

Reply:
We decided to do it because we added several times new parameters to this function and had to change the signature every time (and now we can just extend the struct). But overall I agree that it doesn't really matter...